### PR TITLE
New Board: Dark. Connects to RPi

### PR DESCRIPTION
**Pull request by lostcaggy**
Added new board type Dark. Connects to Raspberry Pi. Details of the ESCAPE board here https://www.kickstarter.com/projects/shrkey/dark-control-robot-controllers-for-the-raspberry-p?token=33e1d674

### CODE REVIEW

**Comment by staroselskii** [comment]
I suggest a rebase on current master and force push afterwards before we start reviewing in details. As of now, it has conflicts.

**Comment by OXINARF** [comment]
Like @staroselskii said you need to rebase, no merge commits are allowed. And you have to follow our commit conventions: http://ardupilot.org/dev/docs/submitting-patches-back-to-master.html

**Comment by lostcaggy** [comment]
I've made some updates and rebased as instructed. Can you check now please
Thanks

File: eclipse.project [hunk 1]
```diff
@@ -1,27 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (magicrub) [inline eclipse.project]: why remove these eclipse files?

File: libraries/AP_HAL_Linux/GPIO.h [hunk 2]
```diff
@@ -25,6 +25,7 @@ class DigitalSource : public AP_HAL::DigitalSource {
       CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RASPILOT || \
       CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_ERLEBRAIN2 || \
       CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BH || \
+	  CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_DARK || \
```
Comment (magicrub) [inline libraries/AP_HAL_Linux/GPIO.h]: whitespace

File: libraries/AP_HAL_Linux/GPIO_RPI.cpp [hunk 3]
```diff
@@ -4,6 +4,7 @@
     CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_RASPILOT || \
     CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_ERLEBRAIN2 || \
     CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BH || \
+	CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_DARK || \
```
Comment (magicrub) [inline libraries/AP_HAL_Linux/GPIO_RPI.cpp]: whitespace

File: libraries/AP_HAL_Linux/HAL_Linux_Class.cpp [hunk 4]
```diff
@@ -136,6 +138,7 @@ static RCInput_AioPRU rcinDriver;
 #elif CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_NAVIO || \
       CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_ERLEBRAIN2 || \
       CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BH || \
+	  CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_DARK || \
```
Comment (magicrub) [inline libraries/AP_HAL_Linux/HAL_Linux_Class.cpp]: please check whitespace and have it match the other files/examples

**Comment by magicrub** [comment]
I still see 4 merges and a couple reverts and lots of whitespace (off by one space). Please clean up and follow the above linked instruction/tutorials.

File: libraries/AP_HAL/AP_HAL_Boards.h [hunk 5]
```diff
@@ -34,6 +34,7 @@
 #define HAL_BOARD_SUBTYPE_LINUX_NAVIO2     1013
 #define HAL_BOARD_SUBTYPE_LINUX_DISCO      1014
 #define HAL_BOARD_SUBTYPE_LINUX_AERO       1015
+#define HAL_BOARD_SUBTYPE_LINUX_DARK	   1016
```
Comment (lostcaggy) [inline libraries/AP_HAL/AP_HAL_Boards.h]: Fixed.

File: eclipse.project (hunk 1 above)
Comment (lostcaggy) [inline eclipse.project]: Added them back in

File: libraries/AP_HAL_Linux/GPIO.h (hunk 2 above)
Comment (lostcaggy) [inline libraries/AP_HAL_Linux/GPIO.h]: fixed

File: libraries/AP_HAL_Linux/GPIO_RPI.cpp (hunk 3 above)
Comment (lostcaggy) [inline libraries/AP_HAL_Linux/GPIO_RPI.cpp]: fixed

File: libraries/SITL/SIM_Calibration.cpp [hunk 6]
```diff
@@ -42,7 +42,7 @@ void SITL::Calibration::update(const struct sitl_input& input)
     } else if (switcher_pwm < 1300) {
         _calibration_poses(rot_accel);
     } else {
-        _angular_velocity_control(input, rot_accel);
+        _attitude_control(input, rot_accel);
```
Comment (lostcaggy) [inline libraries/SITL/SIM_Calibration.cpp]: Not sure but think have now fixed.

File: modules/mavlink [hunk 7]
```diff
@@ -1 +1 @@
-Subproject commit 4e086b51a20c0d09eeb095d012a0fcaccf079d6e
```
Comment (lostcaggy) [inline modules/mavlink]: Hi I've done a rebase. No changes were made to a module apart from updating them. Could you check again please.

**Comment by lostcaggy** [comment]
That's all the whitespace cleaned up. I've done a rebase but not too sure about the GIT side of things. I've tried to follow the tutorials/instructions as much as possible. Thanks

**Comment by magicrub** [comment]
you sure you did a rebase on master?

**Comment by lostcaggy** [comment]
Yes I did

**Comment by magicrub** [comment]
new PR: https://github.com/ArduPilot/ardupilot/pull/5008

**Pull request closed without merging**